be/src/olap/rowset/segment_v2/inverted_index_compaction.cpp

```diff
@@ -57,12 +62,6 @@ Status compact_column(int64_t index_id, std::vector<lucene::store::Directory*>&
     // when index_writer is destroyed, if closeDir is set, dir will be close
     // _CLDECDELETE(dir) will try to ref_cnt--, when it decreases to 1, dir will be destroyed.
     _CLDECDELETE(dir)
-    for (auto* d : src_index_dirs) {
-        if (d != nullptr) {
-            d->close();
```

Comment:
`DorisCompoundReader` should explicitly call the `close()` to release resource.

Reply:
DorisCompoundReader::~DorisCompoundReader() {
    if (!_closed) {
        try {
            close();
        } catch (CLuceneError& err) {
            LOG(ERROR) << "DorisCompoundReader finalize error:" << err.what();
        }
    }
    _CLDELETE(entries)
}